medusa/providers/torrent/json/torrentleech.py

```diff
@@ -0,0 +1,223 @@
+            'details': urljoin(self.url, 'torrent/{id}'),
+        }
+
+        # Proper Strings
+        self.proper_strings = ['PROPER', 'REPACK', 'REAL', 'RERIP']
+
+        # Miscellaneous Options
+        self.max_torrents = 100
+
+        # Torrent Stats
+        self.minseed = None
+        self.minleech = None
+
+        # Cache
+        self.cache = tv.Cache(self)
+
+    def search(self, search_strings, age=0, ep_obj=None, **kwargs):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        # TV/Episodes, TV/BoxSets, TV/Episodes HD, Animation/Anime, Animation/Cartoons, Foreign
+        # 26,          27,         32,             34,              35,                 44
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            # Configured for mode == 'RSS'
+            search_params = {
+                'categories': ['2', '26', '27', '32', '7', '34', '35', '44'],
+                'query': '',
+                'orderby': 'added',
+                'order': 'desc',
+            }
+
+            for search_string in search_strings[mode]:
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+
+                    search_params['categories'] = ['2', '7', '35']
+                    search_params['categories'] += ['26', '32', '44'] if mode == 'Episode' else ['27']
+                    if self.series and self.series.is_anime:
+                        search_params['categories'] += ['34']
+
+                    search_params['query'] = search_string
+                    search_params['orderby'] = 'seeders'
+
+                fragments = 'categories/{categories}/'
+                if search_params['query']:
+                    fragments += 'query/{query}/'
+                fragments += 'orderby/{orderby}/order/{order}/'
+
+                path_params = fragments.format(
+                    categories=','.join(search_params['categories']),
+                    query=search_params['query'],
+                    orderby=search_params['orderby'],
+                    order=search_params['order']
+                )
+                search_url = urljoin(self.urls['search'], path_params)
+
+                json = self.session.get_json(search_url)
+                if not json:
+                    log.debug('No data returned from provider')
+                    continue
+
+                results += self.parse(json, mode)
+
+                # Pagination
+                num_found = json.get('numFound', 0)
+                per_page = json.get('perPage', 35)
```

Comment:
Encourage users to set their site profile settings to 100 results per page (?)
It can't be tweaked in the search url, only on the site.